app/models/concerns/client_image_consumer.rb

```diff
@@ -10,46 +10,43 @@ module ClientImageConsumer
     has_many :client_files unless respond_to?(:client_files)
     has_many :source_eto_client_lookups, through: :source_clients, source: :eto_client_lookups
 
-    # finds an image for the client. there may be more then one available but this
-    # method will select one more or less at random. returns no_image_on_file_image
-    # if none is found. returns that actual image bytes
-    # FIXME: invalidate the cached image if any aspect of the client changes
-    def image(cache_for = 10.minutes) # rubocop:disable Lint/UnusedMethodArgument
-      return nil unless GrdaWarehouse::Config.get(:eto_api_available)
-
-      # Use an uploaded headshot if available
-      faked_image_data = local_client_image_data || fake_client_image_data
-      return faked_image_data unless Rails.env.production?
-
-      # Use the uploaded client image if available, otherwise use the API, if we have access
+    # Returns actual image bytes. Cache time limit not yet implemented.
+    def image(_cache_for = 10.minutes)
```

Comment:
@eanders after staring at the code for a few more minutes, I was wondering whether this method `image` should actually use `image_for_source_client` below as a subroutine? They seem to repeat some of the same logic.

This is basically untested pseudocode but I can fix it up if you think it's a good idea -- something like,
```
def image
  image_data = local_client_image_data
  image_data ||= image_for_source_client
  image_data || self.class.no_image_on_file_image
```

Reply:
I think the reason we avoided that is because `image` is executed against a destination client, so we need to loop over the source clients.  Might be cleaner to replace:
```
 # Check for HMIS uploaded images
       # Queries client_files once for every source client - not ideal, but there are never that many
       image_data ||= source_clients.detect(&:local_hmis_image)&.as_thumb

       # If we still don't have an image at this point, look it up in the ETO API
       image_data ||= source_clients.detect(&:eto_source_image_data)
```
with 
```
image_data ||= source_clients.detect(&:image_for_source_client)
```

I'm not sure if detect could match `''` or if we'd need to return `nil` from `image_for_source_client`.

That does seem somewhat cleaner though, so feel free to take a stab at it!